packages/@aws-cdk/aws-eks-v2-alpha/lib/cluster.ts

```diff
@@ -1055,6 +1125,49 @@ export class Cluster extends ClusterBase {
       ],
     });
 
+    const autoModeEnabled = !(props.defaultCapacityType !== undefined && props.defaultCapacityType !== DefaultCapacityType.AUTOMODE);
+
+    // Throw when using nodePools or nodeRole without using AUTOMODE
+    if (!autoModeEnabled && (props.compute?.nodePools || props.compute?.nodeRole)) {
```

Comment:
can this be simplified by checking `if (!autoModeEnabled && props.compute)`? The error message could be updated as well.

Reply:
makes sense!